File: src/VB.NET/PDFRasterizer/PDFRasterizer/Form1.vb
Public Class Form1

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Me.Icon = New Icon(GetType(Form), "wfc.ico")
        Me.DynamicDotNetTwain1.IfThrowException = False
        Me.cmbPDFResolution.DropDownStyle = ComboBoxStyle.DropDownList
        Me.cmbPDFResolution.Items.Add("100")
        Me.cmbPDFResolution.Items.Add("150")
        Me.cmbPDFResolution.Items.Add("200")
        Me.cmbPDFResolution.Items.Add("300")
        Me.cmbPDFResolution.SelectedIndex = 0

        Me.rdbBMP.Checked = True

        Dim strDllPath As String
        Dim m_strCurrentDirectory As String
        m_strCurrentDirectory = Application.ExecutablePath
        Dim pos As Integer
        pos = m_strCurrentDirectory.LastIndexOf("\Samples\")
        If (pos <> -1) Then
            m_strCurrentDirectory = m_strCurrentDirectory.Substring(0, m_strCurrentDirectory.IndexOf("\", pos)) + "\"
            strDllPath = m_strCurrentDirectory + "Redistributable\PDFResources\"
        Else
            pos = m_strCurrentDirectory.LastIndexOf("\bin\")
            m_strCurrentDirectory = m_strCurrentDirectory.Substring(0, m_strCurrentDirectory.IndexOf("\", pos)) + "\"
            strDllPath = m_strCurrentDirectory + "Redistributable\PDFResources\"
        End If
        Me.DynamicDotNetTwain1.PDFRasterizerDllPath = strDllPath
        Me.DynamicDotNetTwain1.IfShowCancelDialogWhenBarcodeOrOCR = True
    End Sub

    Private Sub btnLoadPDF_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnLoadPDF.Click
        Dim openfiledlg As New OpenFileDialog
        Try
            openfiledlg.Filter = "PDF|*.PDF"
            openfiledlg.FilterIndex = 0
            openfiledlg.Multiselect = True

            Dim strfilename As String
            If (openfiledlg.ShowDialog() = DialogResult.OK) Then
                For Each strfilename In openfiledlg.FileNames
                    Me.DynamicDotNetTwain1.ConvertPDFToImage(strfilename, CInt(cmbPDFResolution.SelectedItem.ToString()))
                Next
            End If
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try      
    End Sub

    Private Sub btnSave_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSave.Click
        Try
            If (Me.DynamicDotNetTwain1.HowManyImagesInBuffer > 0) Then
                Me.LabMsg.Text = ""
                Dim dlgFileSave As New SaveFileDialog
                Dim strFile As String 'The file name use to save the acquired image

                If (rdbBMP.Checked = True) Then
                    dlgFileSave.Filter = "BMP File (*.bmp)|*.bmp"
                ElseIf (rdbJPEG.Checked = True) Then
                    dlgFileSave.Filter = "JPEG File (*.jpg)|*.jpg"
                ElseIf (rdbPNG.Checked = True) Then
                    dlgFileSave.Filter = "PNG File (*.png)|*.png"
                ElseIf (rdbTIFF.Checked = True) Then
                    dlgFileSave.Filter = "TIFF File (*.tif)|*.tif"
                Else
                    dlgFileSave.Filter = "PDF File (*.pdf)|*.pdf"
                End If

                dlgFileSave.InitialDirectory = CurDir()
                dlgFileSave.FileName = ""
                If (dlgFileSave.ShowDialog() = DialogResult.OK) Then
                    strFile = dlgFileSave.FileName
                    If (rdbBMP.Checked = True) Then
                        DynamicDotNetTwain1.SaveAsBMP(strFile, DynamicDotNetTwain1.CurrentImageIndexInBuffer)

                    ElseIf (rdbJPEG.Checked = True) Then
                        DynamicDotNetTwain1.SaveAsJPEG(strFile, DynamicDotNetTwain1.CurrentImageIndexInBuffer)

                    ElseIf (rdbPNG.Checked = True) Then
                        DynamicDotNetTwain1.SaveAsPNG(strFile, DynamicDotNetTwain1.CurrentImageIndexInBuffer)

                    ElseIf (rdbTIFF.Checked = True) Then
                        If (chbMultiPageTIFF.CheckState = 1) Then
                            DynamicDotNetTwain1.SaveAllAsMultiPageTIFF(strFile)
                        Else
                            DynamicDotNetTwain1.SaveAsTIFF(strFile, DynamicDotNetTwain1.CurrentImageIndexInBuffer)
                        End If
                    Else
                        If (chbMultiPagePDF.CheckState = 1) Then
                            DynamicDotNetTwain1.SaveAllAsPDF(strFile)
                        Else
                            DynamicDotNetTwain1.SaveAsPDF(strFile, DynamicDotNetTwain1.CurrentImageIndexInBuffer)
                        End If
                    End If
                End If
            Else
                Me.LabMsg.ForeColor = Color.Red
                Me.LabMsg.Text = "Please load a PDF file first"
                Me.LabMsg.Location = New Point(Me.GroupBox2.Size.Width / 2 - Me.LabMsg.Size.Width / 2, Me.LabMsg.Location.Y)
            End If
        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try  
    End Sub

    Private Sub rdbBMP_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdbBMP.CheckedChanged
        chbMultiPagePDF.Checked = False
        chbMultiPagePDF.Enabled = False
        chbMultiPageTIFF.Checked = False
        chbMultiPageTIFF.Enabled = False
    End Sub

    Private Sub rdbJPEG_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdbJPEG.CheckedChanged
        chbMultiPagePDF.Checked = False
        chbMultiPagePDF.Enabled = False
        chbMultiPageTIFF.Checked = False
        chbMultiPageTIFF.Enabled = False
    End Sub

    Private Sub rdbPNG_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdbPNG.CheckedChanged
        chbMultiPagePDF.Checked = False
        chbMultiPagePDF.Enabled = False
        chbMultiPageTIFF.Checked = False
        chbMultiPageTIFF.Enabled = False
    End Sub

    Private Sub rdbTIFF_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdbTIFF.CheckedChanged
        chbMultiPagePDF.Checked = False
        chbMultiPagePDF.Enabled = False
        chbMultiPageTIFF.Checked = True
        chbMultiPageTIFF.Enabled = True
    End Sub

    Private Sub rdbPDF_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles rdbPDF.CheckedChanged
        chbMultiPagePDF.Checked = True
        chbMultiPagePDF.Enabled = True
        chbMultiPageTIFF.Checked = False
        chbMultiPageTIFF.Enabled = False
    End Sub
End Class

File: src/PDFRasterizerDemo/C#/PDFRasterizer/PDFRasterizer/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace PDFReaderDemo
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            Initialization();
        }

        protected void Initialization()
        {
            this.Icon = new Icon(typeof(Form), "wfc.ico");
            dynamicDotNetTwain1.IfThrowException = false;
            this.cmbPDFResolution.DropDownStyle = ComboBoxStyle.DropDownList;
            this.cmbPDFResolution.Items.Add("100");
            this.cmbPDFResolution.Items.Add("150");
            this.cmbPDFResolution.Items.Add("200");
            this.cmbPDFResolution.Items.Add("300");
            this.cmbPDFResolution.SelectedIndex = 0;

            this.rdbBMP.Checked = true;
            
            string strDllFolder = Application.ExecutablePath;
            strDllFolder = strDllFolder.Replace("/", "\\");
            int pos = strDllFolder.LastIndexOf("\\bin\\");
            if (pos != -1)
            {
                strDllFolder = strDllFolder.Substring(0, strDllFolder.IndexOf(@"\", pos)) + @"\Redistributable\PDFResources\";
            }
            else
            {
                pos = strDllFolder.LastIndexOf("\\");
                strDllFolder = strDllFolder.Substring(0, strDllFolder.IndexOf(@"\", pos)) + @"\";
            }

            dynamicDotNetTwain1.PDFRasterizerDllPath = strDllFolder;
            dynamicDotNetTwain1.IfShowCancelDialogWhenBarcodeOrOCR = true;
        }

        private void btnLoadPDF_Click(object sender, EventArgs e)
        {
            try
            {
                OpenFileDialog openfiledlg = new OpenFileDialog();
                openfiledlg.Filter = "PDF|*.PDF";
                openfiledlg.FilterIndex = 0;
                openfiledlg.Multiselect = true;

                if (openfiledlg.ShowDialog() == DialogResult.OK)
                {
                    foreach (string strfilename in openfiledlg.FileNames)
                    {
                        this.dynamicDotNetTwain1.ConvertPDFToImage(strfilename, float.Parse(cmbPDFResolution.SelectedItem.ToString()));
                    }
                }
            }
            catch(Exception exp)
            {
                MessageBox.Show(exp.Message);
            }
        }

        private void btnSave_Click(object sender, EventArgs e)
        {
            try
            {
                if (dynamicDotNetTwain1.HowManyImagesInBuffer > 0)
                {
                    this.labMsg.Text = "";
                    SaveFileDialog savefdlg = new SaveFileDialog();
                    savefdlg.InitialDirectory = System.IO.Directory.GetCurrentDirectory();
                    savefdlg.FileName = "";

                    if (rdbBMP.Checked == true)
                    {
                        savefdlg.Filter = "BMP File(*.bmp)| *.bmp";

                    }

                    if (rdbJPEG.Checked == true)
                    {
                        savefdlg.Filter = "JPEG File(*.jpeg)| *.jpeg";

                    }

                    if (rdbPNG.Checked == true)
                    {
                        savefdlg.Filter = "PNG File(*.png) | *.png";

                    }

                    if (rdbTIFF.Checked == true)
                    {
                        savefdlg.Filter = "TIFF File(*.tiff)| *.tiff";

                    }

                    if (rdbPDF.Checked == true)
                    {
                        savefdlg.Filter = "PDF File(*.pdf)| *.pdf";

                    }

                    if (savefdlg.ShowDialog() == DialogResult.OK)
                    {

                        string strFilename = savefdlg.FileName;

                        if (rdbBMP.Checked == true)
                        {
                            this.dynamicDotNetTwain1.SaveAsBMP(strFilename, this.dynamicDotNetTwain1.CurrentImageIndexInBuffer);
                        }

                        if (rdbJPEG.Checked == true)
                        {
                            this.dynamicDotNetTwain1.SaveAsJPEG(strFilename, this.dynamicDotNetTwain1.CurrentImageIndexInBuffer);
                        }

                        if (rdbPNG.Checked == true)
                        {
                            this.dynamicDotNetTwain1.SaveAsPNG(strFilename, this.dynamicDotNetTwain1.CurrentImageIndexInBuffer);
                        }

                        if (rdbTIFF.Checked == true)
                        {
                            if (chbMultiPageTIFF.Checked == true)
                            {
                                this.dynamicDotNetTwain1.SaveAllAsMultiPageTIFF(strFilename);
                            }
                            else
                                this.dynamicDotNetTwain1.SaveAsTIFF(strFilename, dynamicDotNetTwain1.CurrentImageIndexInBuffer);
                        }

                        if (rdbPDF.Checked == true)
                        {
                            if (chbMultiPagePDF.Checked == true)
                                dynamicDotNetTwain1.SaveAllAsPDF(strFilename);
                            else
                                dynamicDotNetTwain1.SaveAsPDF(strFilename, dynamicDotNetTwain1.CurrentImageIndexInBuffer);
                        }
                    }
                }
                else
                {
                    this.labMsg.ForeColor = Color.Red;
                    this.labMsg.Text = "Please load a PDF file first";
                    this.labMsg.Location = new Point(this.groupBox2.Size.Width / 2 - this.labMsg.Size.Width / 2, this.labMsg.Location.Y);
                }
            }
            catch (Exception exp)
            {
                MessageBox.Show(exp.Message);
            }
        }

        private void rdbBMP_CheckedChanged(object sender, EventArgs e)
        {
            chbMultiPagePDF.Checked = false;
            chbMultiPagePDF.Enabled = false;
            chbMultiPageTIFF.Checked = false;
            chbMultiPageTIFF.Enabled = false;
        }

        private void rdbJPEG_CheckedChanged(object sender, EventArgs e)
        {
            chbMultiPagePDF.Checked = false;
            chbMultiPagePDF.Enabled = false;
            chbMultiPageTIFF.Checked = false;
            chbMultiPageTIFF.Enabled = false;
        }

        private void rdbPNG_CheckedChanged(object sender, EventArgs e)
        {
            chbMultiPagePDF.Checked = false;
            chbMultiPagePDF.Enabled = false;
            chbMultiPageTIFF.Checked = false;
            chbMultiPageTIFF.Enabled = false;
        }

        private void rdbTIFF_CheckedChanged(object sender, EventArgs e)
        {
            chbMultiPagePDF.Checked = false;
            chbMultiPagePDF.Enabled = false;
            chbMultiPageTIFF.Checked = true;
            chbMultiPageTIFF.Enabled = true;
        }

        private void rdbPDF_CheckedChanged(object sender, EventArgs e)
        {
            chbMultiPagePDF.Checked = true;
            chbMultiPagePDF.Enabled = true;
            chbMultiPageTIFF.Checked = false;
            chbMultiPageTIFF.Enabled = false;
        }

    }
}
